ntbs-service/Jobs/ReportingDataRefreshJob.cs

```diff
@@ -1,123 +1,51 @@
-        protected readonly string _specimenMatchingConnectionString = "";
-        protected readonly string _migrationConnectionString = "";
+        private readonly IExternalStoredProcedureRepository _externalStoredProcedureRepository;
 
-        public ReportingDataRefreshJob(IConfiguration configuration)
-        : base(configuration)
+        public ReportingDataRefreshJob(IExternalStoredProcedureRepository externalStoredProcedureRepository)
         {
-            _reportingDatabaseConnectionString = _configuration.GetConnectionString(Constants.DbConnectionStringReporting);
-            _specimenMatchingConnectionString = _configuration.GetConnectionString(Constants.DbConnectionStringSpecimenMatching);
-            _migrationConnectionString = _configuration.GetConnectionString(Constants.DbConnectionStringMigration);
-
-            _sqlString = "";
-            _parameters = null;
+            _externalStoredProcedureRepository = externalStoredProcedureRepository;
         }
 
-        /// PerformContext context is passed in via Hangfire Server
-        public override async Task Run(PerformContext context, IJobCancellationToken token)
+        // PerformContext context is passed in via Hangfire Server
+        public async Task Run(PerformContext context)
         {
-            _context = context;
-            Log.Information($"Starting reporting data refresh job.");
+            LogInfo(context, "Starting reporting data refresh job");
 
             try
             {
-                var stepOneResults = await ExecuteSpecimenMatchingGenerateStoredProcedure(token);
-                var stepTwoResults = await ExecuteReportingGenerateStoredProcedure(token);
-                var stepThreeResults = await ExecuteMigrationGenerateStoredProcedure(token);
-
-                var allResults = new List<dynamic>();
-                allResults.AddRange(stepOneResults);
-                allResults.AddRange(stepTwoResults);
-                allResults.AddRange(stepThreeResults);
-
-                var success = DidExecuteSuccessfully(allResults);
+                LogInfo(context, "Starting specimen-matching uspGenerate");
+                var stepOneResults =
+                    await _externalStoredProcedureRepository.ExecuteSpecimenMatchingGenerateStoredProcedure();
+                CheckExecutedSuccessfully(context, stepOneResults);
+
+                LogInfo(context, "Starting reporting uspGenerate");
+                var stepTwoResults =
+                    await _externalStoredProcedureRepository.ExecuteReportingGenerateStoredProcedure();
+                CheckExecutedSuccessfully(context, stepTwoResults);
+
+                LogInfo(context, "Starting migration uspGenerate");
+                var stepThreeResults =
+                    await _externalStoredProcedureRepository.ExecuteMigrationGenerateStoredProcedure();
+                CheckExecutedSuccessfully(context, stepThreeResults);
```

Comment:
So I've tweaked the way we proceed with errors here to make it consistent, but would like to run it past you:

Before, in short, depending how a previous stored proc failed, we might or might not proceed to the next stored proc.
This was because if the first stored proc threw an exception in the C# SQL library (eg. if the connection was refused, the stored proc didn't exist, etc.) then we would not execute any of the following stored procs. However, if the stored proc ran but had errors (and we have a try/catch in the SQL, so an issue with one of the inner stored procs results in an error message string getting passed back up. but no C# exception), then we would continue to run the subsequent stored procs. 

To make this consistent this now checks to see if there are any SQL errors after each stored proc (as well as throwing up C# exceptions), and does not run the next one if the previous one failed at either level, in C# or in SQL.

Reply:
I think this is a good change :-)